### PR TITLE
GPG signature verification support

### PR DESCRIPTION
**Pull request by xen0l**
This PR adds support for verifying GPG signatures for component downloads. This increases security. I did following changes:
- made userland-fetch PEP8 compliant and code should be more readable
- ported the functionality from userland-gate and adapted to oi-userland
- added COMPONENT_SIG_URL for some components

Workflow is rather simple:
- add COMPONENT_SIG_URL and run gmake download. Usually, these suffixes are standard:
  **asc**, **sign** or **sig**.
- check if corresponding public key has been imported into pubring.gpg:

```
gpg2 --homedir=$(git rev-parse --show-toplevel)/tools/.gnupg --list-keys
```
- commit Makefile change and $(git rev-parse --show-toplevel)/tools/.gnupg/pubring.gpg

With this functionality it is possible to omit component tarball checksum and use GPG verification for downloaded component. We should add it where possible once merged.

### CODE REVIEW

File: tools/userland-fetch [hunk 1]
```diff
@@ -28,246 +28,371 @@
...
+    # configuration and keyring.
+    proc = subprocess.Popen(["git", "rev-parse", "--show-toplevel"], stdout=subprocess.PIPE)
+    proc.wait()
+    if proc.returncode != 0:
+        return False
+    out, err = proc.communicate()
+    gpgdir = os.path.join(out.strip(), "tools", ".gnupg")
+
+    # Skip the permissions warning: none of the information here is private,
+    # so not having to worry about getting mercurial keeping the directory
```
Comment (pyhalov) [inline tools/userland-fetch]: s/mercurial/git/ :)

Comment (xen0l) [reply tools/userland-fetch]: Fixed.

File: tools/userland-fetch [hunk 2]
```diff
@@ -28,246 +28,371 @@
...
+import subprocess
+import re
+import gzip
+import bz2
 from urllib import splittype
 from urllib2 import urlopen
 from urllib2 import Request
 import hashlib
 
+
```
Comment (pyhalov) [inline tools/userland-fetch:44]: please, clean up this new line

File: tools/userland-fetch [hunk 3]
```diff
@@ -28,246 +28,371 @@
...
+        yield name
 
-	return urls
 
 def usage():
-	print "Usage: %s [-a|--user-agent (user-agent)] [-f|--file (file)] " \
-		"[-l|--link] [-h|--hash (hash)] [-s|--search (search-dir)] " \
-		"-u|--url (url)" % (sys.argv[0].split('/')[-1])
-	sys.exit(1)
+    print "Usage: %s [-f|--file (file)] [-l|--link] [-h|--hash (hash)] " \
```
Comment (pyhalov) [inline tools/userland-fetch]: where -a has gone?

Comment (xen0l) [reply tools/userland-fetch]: Oops, I will add it back.

File: tools/userland-fetch (hunk 2 above)
Comment (xen0l) [reply tools/userland-fetch:44]: This is PEP8 specific, so I would rather leave it there. (PEP8 mandates to use two blank lines between imports and function/class definitions).

File: tools/userland-fetch [hunk 4]
```diff
@@ -28,246 +28,371 @@
...
+                # the target file if the signature was valid.
+                if not sig_valid:
+                    try:
+                        os.remove(name)
+                    except OSError:
+                        pass
+
+                    continue
+
+        sys.exit(0)
```
Comment (pyhalov) [inline tools/userland-fetch]: what if there's no signature and hash invalid? Will this line be reached (hard to follow intendation :)

File: tools/userland-fetch (hunk 2 above)
Comment (pyhalov) [reply tools/userland-fetch:44]: OK
